extra-examples/CustomPayload2/Function.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using CloudNative.CloudEvents;
+using CloudNative.CloudEvents.NewtonsoftJson;
+using Google.Cloud.Functions.Framework;
+using Google.Cloud.Functions.Hosting;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using ThirdPartyPackage;
+
+namespace CustomPayload2
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void ConfigureServices(WebHostBuilderContext context, IServiceCollection services) =>
```

Comment:
It is a bit odd to me that the formatter is configured as a Functions Framework service and not on the Google Events library. Said another way: how do we expect customers will use the Google Events library outside of the Functions Framework? For example consider the use case of deploying an ASP.NET app to CloudRun and point an EventArc subscription at it. I would expect to be able to do something like:


```dotnet
// using stuff
var builder = WebApplication.CreateBuilder(args);

// I guess would need to configure a service here?

var app = builder.Build();

app.MapPost("/", (CloudEvent cloudEvent) => { 
  // now I want to work with a Google Event data payload...
});
```

Reply:
This is in a scenario where you're not using the Google Events library at all - it's not a Google event, it's a custom event payload.

Using the Google Events library outside the Functions Framework, you'd need to set up code to receive the CloudEvent - that's one area of the SDK that's still slightly TBD, but doesn't feel like it's in the remit of the Google Events library particularly - that should make it easy to parse the payload as a Google event payload, but that's all. (If you're writing a web server to handle a non-Google-payload CloudEvent request, it's not really our business, even if it happens that the request is made by Eventarc.)